### PR TITLE
Replace nevow with twisted.web.template in status.RetrieveStatusPage

### PR DESCRIPTION
**Pull request by sajith**
Fixes [3290](https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3290).

Screenshot of the page after the port:

wui-retrieve-after

----

Before:

wui-retrieve-before

### CODE REVIEW

File: src/allmydata/web/retrieve-status.xhtml [hunk 1]
```diff
@@ -1,43 +1,48 @@
...
+    <h2>Retrieve Results</h2>
+
     <ul>
-      <li>Fetching: <span n:render="time" n:data="time_fetch" />
-      (<span n:render="rate" n:data="rate_fetch" />)</li>
-      <li>Decoding: <span n:render="time" n:data="time_decode" />
-      (<span n:render="rate" n:data="rate_decode" />)</li>
-      <li>Decrypting: <span n:render="time" n:data="time_decrypt" />
-      (<span n:render="rate" n:data="rate_decrypt" />)</li>
+      <li t:render="encoding" />
```
Comment (meejah) [inline src/allmydata/web/retrieve-status.xhtml:26]: No action required, but re-indenting this does make it a little harder to review.

File: src/allmydata/web/status.py [hunk 2]
```diff
@@ -673,90 +673,129 @@ def render_progress(self, ctx, data):
...
-    docFactory = getxmlfile("retrieve-status.xhtml")
 
-    def __init__(self, data):
-        rend.Page.__init__(self, data)
-        self.retrieve_status = data
 
-    def render_started(self, ctx, data):
-        started_s = render_time(data.get_started())
-        return started_s
+# Renders "/status/retrieve-%d".
```
Comment (meejah) [inline src/allmydata/web/status.py:678]: docstring instead of comment would be better

File: src/allmydata/web/status.py [hunk 3]
```diff
@@ -673,90 +673,129 @@ def render_progress(self, ctx, data):
...
+class RetrieveStatusPage(MultiFormatResource):
 
-    def render_si(self, ctx, data):
-        si_s = base32.b2a_or_none(data.get_storage_index())
+    def __init__(self, retrieve_status):
+        """
+        :retrieve_status retrieve.RetrieveStatus: stats provider.
```
Comment (meejah) [inline src/allmydata/web/status.py:683]: needs `param` (and the type is first) .. `:param <optional type> arg_name: the description`

File: src/allmydata/web/status.py [hunk 4]
```diff
@@ -673,90 +673,129 @@ def render_progress(self, ctx, data):
...
 
-    def render_problems(self, ctx, data):
-        problems = data.get_problems()
+    @renderer
+    def problems(self, req, tag):
+        problems = self._retrieve_status.get_problems()
         if not problems:
             return ""
-        l = T.ul()
+        l = tags.ul()
```
Comment (meejah) [inline src/allmydata/web/status.py:745]: maybe at least `ul`? `l` can be confused with `1` easily for many typefaces

**Review by meejah — APPROVED** [review]
couple inline comments, overall looks good.

File: src/allmydata/web/status.py (hunk 2 above)
Comment (sajith) [reply src/allmydata/web/status.py]: Made the change in 62760c0.

File: src/allmydata/web/status.py (hunk 3 above)
Comment (sajith) [reply src/allmydata/web/status.py]: Made this change in 812f039.

File: src/allmydata/web/status.py (hunk 4 above)
Comment (sajith) [reply src/allmydata/web/status.py]: Renamed `l` to `ul` in 5aa0dd4.

File: src/allmydata/web/status.py (hunk 3 above)
Comment (sajith) [reply src/allmydata/web/status.py]: Made another change in 9d5d338.  I missed the `:param <type> <name>` format, but then so did `pydocstyle`...

**Pull request merged**